src/us/kbase/typedobj/db/MongoTypeStorage.java

```diff
@@ -375,10 +375,11 @@ public String getModuleSpecRecord(String moduleName, long version)
 			final DBObject whereCondition,
 			final BasicDBObject projection,
 			final Class<T> clazz) {
-		final DBCursor find = col.find(whereCondition, projection);
 		final List<T> data = new LinkedList<>();
-		for (final DBObject dbo: find) {
-			data.add(toObj(dbo, clazz));
+		try (final DBCursor find = col.find(whereCondition, projection)) {
```

Comment:
This `try()` format is the equivalent of the `with` statement in python. In this case it ensures that the `DBCursor` is always closed when the block exists. There are several of these changes in this PR.